### DISCUSSION TITLE
Projects (Beta) - View Archived cards

### DISCUSSION THREAD

**Discussion by forgedhallpass** [community/community#7501] [Projects and Issues] [answered]
Is there a way to view archived items in a beta project through the UI, or the API? I can't seem to find it...

**Comment by tmspzz** [comment]
I have found this guide https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards/archiving-cards-on-a-project-board but it only works for the non-beta projects.

It would be great to have it for the new projects too. Currently there is no way that I know of to surface archived cards

**Comment by mobile-sergey** [comment]
On old project we can view archives on addresses:
https://github.com/{UserName}/{RepositoryName}/projects/{ProjectNumber}/cards/archived/ - user repository project
https://github.com/orgs/{OrganizationName}/{RepositoryName}/projects/{ProjectNumber}/cards/archived/ - org repository project
https://github.com/orgs/{OrganizationName}/projects/{ProjectNumber}/cards/archived/ - org project
But in [Project Beta] I can't found address of archived cards. All addressed that I check are show 404 error for me
(for example: 
https://github.com/orgs/{OrganizationName}/projects/{ProjectNumber}/cards/archived/ 
https://github.com/orgs/{OrganizationName}/projects/{ProjectNumber}/views/archived/ 
https://github.com/orgs/{OrganizationName}/projects/{ProjectNumber}/views/cards/archived/ 
https://github.com/orgs/{OrganizationName}/projects/{ProjectNumber}/views/{ViewNumber}/cards/archived/ 
All get 404 error)

**Reply by forgedhallpass** [reply]
I've even tried retrieving the information through the GraphQL API, but it doesn't seem to be made available there either.

https://docs.github.com/en/graphql/overview/explorer

**Reply by mobile-sergey** [reply]
For GraphQL API I need to Organization access, but I haven't it (because I'm just developer in company, not owner)
But this is good decision if it worked for [Projects Beta]

**Reply by leitneratselerity** [reply]
This seems to have deeper implications: Beta fields for issues are only visible (also the the GraphQL API!) if the issue is not archived. If you track metrics via those Beta fields (e.g., time spent on the issue) you therefore cannot archive items from Beta. Our current work-around is to have a special Archive State, to avoid that our Done column fills up (instead, we have a crammed Archive column...)

**Reply by IanJones4** [reply]
There is an isArchived flag listed under [ProjectNextItem](https://docs.github.com/en/graphql/reference/objects#projectnextitem).

Viewing the items from the Project doesn't show the Archived item.

````graphql
{
  organization(login: "<ORG_NAME>") {
    projectNext(number: <PROJECT_NUMBER>) {
      items(first: 100) {
        pageInfo {
          hasNextPage
          endCursor
        }
        edges {
          node {
            content {
              ... on Issue {
                id
                title
                number
              }
            }
            isArchived
          }
        }
      }
    }
  }
}
````

But when querying from the Repository to show the ProjectNextItem does show the archived status:

````graphql
{
  organization(login: "<ORG_NAME>") {
  	repository(name:"<REPO_NAME>"){
      issue(number:<ISSUE_NUMBER>){
        number
        id
        title
        projectNextItems(includeArchived:true, first:100){
          edges{
            node{
              isArchived
            }            
          }
        }
      }
    }
  }
}
````

It would be good to have the "includeArchived" parameter under ProjectNext.

**Comment by andre-wells** [comment]
In the current Project Boards, there is an easy way to [view archived items and to restore them](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards/archiving-cards-on-a-project-board).

Maybe we just need a menu option similar next to Workflows and Settings?

![image](https://user-images.githubusercontent.com/66315105/143864619-e6757112-3b0d-47cd-bfd5-cd7a68232b05.png)

Currently the only way to restore an item is to go to the item itself (if I can remember which one I'm looking for)

![image](https://user-images.githubusercontent.com/66315105/143866739-541ef1ec-5de1-451b-a341-18c2097cfb7e.png)

**Reply by mobile-sergey** [reply]
[Restoring cards on a project board from the sidebar](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards/archiving-cards-on-a-project-board#restoring-cards-on-a-project-board-from-the-sidebar) - this is only about projects, not about [Projects Beta]

Yes, we just need a menu option similar next to Workflows and Settings.

Thanks for telling about restore from item card. This is only one way to restore card in [Project Beta] may be useful.

**Reply by forgedhallpass** [reply]
If you track the backlog and sprints of an organization using Projects Beta, and you've archived the tickets from a finished sprint, but forgot to calculate the sprint velocity, you are in for a treat. I've managed to make this mistake, and even though you can use the sidebar to restore items, you have to backtrack all the items that were assigned to a project (having multiple active repositories makes it even more fun, because it can easily add up to hundreds of tickets) and restore them all, filter them again, and finally do your calculations/processing, before re-archiving. Even with some partial automation, it was quite painful.

I've ended up creating a bookmarklet, which does the calculations for me and then exports the contents of a table view to CSV which I store in the cloud for historical purposes.

**Reply by mobile-sergey** [reply]
I'm happy that you found decision for you case.
My case easier: After archiving elements we need to see all list of archiving elements. 
For example it needed for analyze or create list of fixes and improvements

**Reply by forgedhallpass** [reply]
You can also mark those tickets with a label before archiving, or do as I did...export them to CSV using workaround (because there is no such feature either), and when needed do filtering using your own automation, or using Google Sheets/Excel.

**Reply by mobile-sergey** [reply]
How to export archived items?

**Comment by azenMatt** [comment]
Hi everyone,

Exploring archived cards is something we are currently working on, we are looking to start with a simple list or archived items with:

- [ ] Ability to restore an item back to the project 
- [ ] Understand **when** it was archived and by **who** 

Let me know if you have any thoughts / feedback on if this would improve your experience with the archive 

![image](https://user-images.githubusercontent.com/7584089/153186531-8c8445ba-5125-4bff-b58d-897a53c4e46e.png)

**Reply by forgedhallpass** [reply]
It would be great to be able to filter them based on multiple criteria, like title, labels, milestone etc.

**Reply by mobile-sergey** [reply]
Yes, I need list of archived items and ability to restore it - like you show on this screenshot.

I am archive all finished items from iteration column, when this iteration was finished. 
It will be useful to see what column this item from.

The best solution if we can see in "Archive" all fields of items (with ability to customize what fields to show)
Because I use 1 Project [Beta] for mobile development in 3 repositories (one for iOS, second for Android, last for Web/CrossPlatform) And some tasks may have same names, but different values of other fields.

Or as wrote forgedhallpass: filter on multiple criteria is good idea, if is good idea too.

**Reply by GavT** [reply]
Awesome to hear this is in the pipeline.

For delivery estimations capturing the card / issue throughput (amount of work delivered over a specific period) and cycle time (the time that the team works on the card / issue until it’s delivered) are essential metrics. It would be even more awesome if the Insights section also used this information to create throughput histograms.

**Reply by jeffadavidson** [reply]
@azenMatt is there any timeline on releasing even this rough version?

**Reply by azenMatt** [reply]
Sooner rather than later - we should start writing code for it in the next week or two

**Reply by mjay617** [reply]
I'm also doing some analysis to find continuous improvement opportunities. So I'd like to be able to see metrics others have already listed as well as a log of status changes with dates, and ability to view and sort archived cards by time period, e.g. per month.

**Reply by mmitoraj** [reply]
It would be really awesome to be able to see the archived cards (both issues and drafts) and filter them by fields. For example, in my team's  Beta Project we use the `Sprint number` filed to be able to track what was done in a particular time period. But unfortunately, without the ability to see the archived cards it won't work.

**Reply by esc** [reply]
this would be useulf

**Comment by dickshaydle** [comment]
I can't find a way to restore the archived cards. Currently it is more like deleting with a "archiving" label. 

Imagine a paper document folder with the label "Archive", that is connected to the paper shredder.

**Reply by forgedhallpass** [reply]
They can be restored, but you need to track down every ticket you'd like to restore separately (which is far from being user friendly). If you identify and open an archived issue, there will be a restore button on the right. See the screenshot from @andre-wells above.

**Reply by dickshaydle** [reply]
If you created an issue already then this is possible. 
But in my case, it was a **draft**. There is no URL / address to a draft that i could find. So its gone forever as it seems. Or at least up to the point when the UI supports this feature.

**Comment by mariorod** [comment, marked as answer]
We have now shipped this functionality 🎉 . See announcement in this changelog: https://github.blog/changelog/2022-04-07-the-new-github-issues-april-7th-update/

**Reply by sicknarlo** [reply]
Any update on if/when this will be included in the GraphQL API?

**Reply by MarkOSullivan94** [reply]
It would be great if we could have the option to include archived items in our boards and table views
